avr-hal-generic/src/i2c.rs

```diff
@@ -388,6 +412,42 @@ impl<H, I2C: I2cOps<H, SDA, SCL>, SDA, SCL, CLOCK> embedded_hal_v0::blocking::i2
     }
 }
 
+impl<H, I2C: I2cOps<H, SDA, SCL>, SDA, SCL, CLOCK> embedded_hal::i2c::I2c<SevenBitAddress>
+    for I2c<H, I2C, SDA, SCL, CLOCK>
+{
+    fn transaction(
+        &mut self,
+        address: u8,
+        operations: &mut [embedded_hal::i2c::Operation<'_>],
+    ) -> Result<(), Self::Error> {
+        let mut previous_direction = Direction::Read;
+        let nb_operations = operations.len();
+        for (idx, operation) in operations.iter_mut().enumerate() {
+            match operation {
+                embedded_hal::i2c::Operation::Read(buffer) => {
+                    if idx == 0 || previous_direction != Direction::Read {
+                        self.p.raw_start(address, Direction::Read)?;
+                    }
+                    self.p.raw_read(buffer)?;
+                    previous_direction = Direction::Read;
+                }
+                embedded_hal::i2c::Operation::Write(bytes) => {
+                    if idx == 0 || previous_direction != Direction::Write {
+                        self.p.raw_start(address, Direction::Write)?;
+                    }
+                    self.p.raw_write(bytes)?;
+                    previous_direction = Direction::Write;
+                }
+            }
+        }
+        if nb_operations > 0 {
```

Comment:
I think it's cleaner to drop the `nb_operations` variable and directly do

```rust
if operations.len() > 0 {
```

here.

Reply:
Hi,
I made this little change.